install_config/router/default_haproxy_router.adoc

```diff
@@ -1500,109 +1500,99 @@ add routes across the namespaces.
-router metrics for consumption by external metrics collection and aggregation
-systems (e.g. Prometheus, statsd).
-
-Depending on your router implementation, the image is appropriately set up and
-the metrics sidecar container is started when the router is deployed. For
-example, the HAProxy-based router implementation defaults to using the
-`prom/haproxy-exporter` image to run as a sidecar container, which can then be
-used as a metrics datasource by the Prometheus server.
-
-[NOTE]
-====
-The `--metrics-image` option overrides the defaults for HAProxy-based router
-implementations and, in the case of custom implementations, enables the image to
-use for a custom metrics exporter or publisher.
-====
-
-ifdef::openshift-enterprise[]
-. Grab the HAProxy Prometheus exporter image from the Docker registry:
-+
-====
-----
-$ sudo docker pull prom/haproxy-exporter
-----
-====
+The
+xref:../../architecture/networking/haproxy-router.adoc#haproxy-metrics[HAProxy router metrics]
+are, by default, exposed or published in
+link:https://prometheus.io/docs/concepts/data_model/[Prometheus format]
+for consumption by external metrics collection and aggregation systems (e.g. Prometheus, statsd)..
+Metrics are also available dirctly from the
+link:https://cbonte.github.io/haproxy-dconv/1.5/configuration.html#9[HAProxy router] in its own CSV format.
 
-. Create the {product-title} router:
-+
+When you create a router, as below,
 ====
 ----
-$ oc adm router --service-account=router --expose-metrics
+$ oc adm router --service-account=router
 ----
 ====
-+
-Or, optionally, use the `--metrics-image` option to override the HAProxy
-defaults:
-+
+metrics are automatically available in prometheus format on the stats-port, default 1936. To suppress metrics collection,
 ====
 ----
-$ oc adm router --service-account=router --expose-metrics \
-    --metrics-image=prom/haproxy-exporter
+$ oc adm router --service-account=router --stats-port=0
 ----
 ====
-endif::[]
-ifdef::openshift-origin[]
-. Grab the HAProxy Prometheus exporter image from the Docker registry:
-+
+
+To switch to the HAProxy CSV format metrics, edit the
+architecture/networking/routes.adoc
+xref:../../architecture/networking/routes.adoc#env-variables[environment variables]
+for the router dc and delete the following lines:
+
 ====
 ----
-$ sudo docker pull prom/haproxy-exporter
+          - name: ROUTER_LISTEN_ADDR
+            value: 0.0.0.0:1936
+          - name: ROUTER_METRICS_TYPE
+            value: haproxy
 ----
 ====
+Where 1936 is the STATS_PORT value.
 
-. Create the {product-title} router:
-+
+[NOTE]
 ====
-----
-$ oc adm router --service-account=router --expose-metrics
-----
+The `--expose-metrics` and `--metrics-image` options are deprecated. The haproxy-exporter
+side car is now integrated into the router controller so you can delete the sidecar from existing
+router DCs. You can continue to use the side car in existing routers. New routers use the integrated metrics.
```

Comment:
Elsewhere we refer to `sidecar container` and not just `sidecar`. Is this the same concept? Maybe use `sidecar container` here?
https://docs.openshift.org/latest/install_config/router/default_haproxy_router.html#exposing-the-router-metrics

Reply:
Changed